src/pudl/settings.py

```diff
@@ -348,6 +348,88 @@ class GlueSettings(FrozenBaseModel):
     ferc1: bool = True
 
 
+@unique
+class GPRATKTechType(StrEnum):
+    """Enum to constrain GridPath RA Toolkit technology types."""
+
+    WIND = auto()
+    SOLAR = auto()
+    # Not yet implemented
+    # THERMAL = auto()
+
+
+@unique
+class GPRATKProcLevel(StrEnum):
+    """Enum to constraint GridPath RA Toolkit processing levels."""
+
+    EXTENDED = auto()
+    # Not yet implemented
+    # AGGREGATED = auto()
+    # ORIGINAL = auto()
+
+
+class GridPathRAToolkitSettings(GenericDatasetSettings):
+    """An immutable pydantic model to validate GridPath RA Toolkit settings."""
+
+    data_source: ClassVar[DataSource] = DataSource.from_id("gridpathratoolkit")
+    technology_types: list[str] = ["wind", "solar"]
+    processing_levels: list[str] = ["extended"]
+    daily_weather: bool = True
+    parts: list[str] = []
+
+    @field_validator("technology_types", "processing_levels")
+    @classmethod
+    def deduplicate_list(cls, v):
+        """Deduplicate technology type and processing level values."""
+        return list(set(v))
+
+    @field_validator("technology_types")
+    @classmethod
+    def allowed_technology_types(cls, v: list[str]) -> list[str]:
+        """Ensure that technology types are valid."""
+        for tech_type in v:
+            if tech_type not in GPRATKTechType:
+                raise ValueError(f"{tech_type} is not a valid technology type.")
+        return v
+
+    @field_validator("processing_levels")
+    @classmethod
+    def allowed_processing_levels(cls, v: list[str]) -> list[str]:
+        """Ensure that processing levels are valid."""
+        for proc_level in v:
+            if proc_level not in GPRATKProcLevel:
+                raise ValueError(f"{proc_level} is not a valid processing level.")
+        return v
+
+    @field_validator("parts")
+    @classmethod
+    def compile_parts(cls, parts: list[str], info: ValidationInfo) -> list[str]:
+        """Based on technology types and processing levels, compile a list of parts."""
+        if info.data["daily_weather"]:
+            parts.append("daily_weather")
+        if (
+            "solar" in info.data["technology_types"]
+            and "extended" in info.data["processing_levels"]
+        ):
+            parts.append("aggregated_extended_solar_capacity")
+        if (
+            "wind" in info.data["technology_types"]
+            and "extended" in info.data["processing_levels"]
+        ):
+            parts.append("aggregated_extended_wind_capacity")
+        if "solar" in info.data["technology_types"] and (
+            "extended" in info.data["processing_levels"]
+            or "aggregated" in info.data["processing_levels"]
+        ):
+            parts.append("solar_capacity_aggregations")
+        if "wind" in info.data["technology_types"] and (
+            "extended" in info.data["processing_levels"]
+            or "aggregated" in info.data["processing_levels"]
+        ):
+            parts.append("wind_capacity_aggregations")
```

Comment:
this is not blocking I think it would be good to condense this logic a little bit so we don't need to add a new if block for every new technology and/or processing level. it might only be worth is as we add more but i think it'd be helpful and i'd be happy to suggest something if you'd like

Reply:
If you've got a clear suggestion sure!  Unfortunately the various combinations of technology & processing don't behave uniformly across the board.  E.g. there's no aggregated version of the thermal derates, but there is an extended version (it's just every single generator gets an extended timeseries), so it jumps straight from original to extended.  Are you thinking having 2 layers of nesting, one for tech type and one for processing level?

Reply:
Okay, gonna punt this until we add more partitions.